packages/api/src/objectSet/ObjectSet.ts

```diff
@@ -245,4 +247,24 @@ export interface ObjectSet<
     properties: Array<P>,
     listener: EXPERIMENTAL_ObjectSetListener<Q, P>,
   ) => { unsubscribe: () => void };
+
+  readonly derive: <
```

Comment:
I'm not sure of the best way to gate this. I want this to be on object set since I want to ensure you can do stuff to an object set before it. Is it crazy to leave this ungated on beta(only merged when e2e works ofc)